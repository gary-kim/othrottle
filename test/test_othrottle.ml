open! Core
open Stdio

(* Config with existing config *)
let () =
  let () =
    match Othrottle.Config.t_from_filepath "../../../test/samples/config.toml" with
    | Ok config -> Othrottle.Config.sexp_of_t config |> Sexp.to_string |> print_endline
    | Error err -> print_endline err
  in
  ()
;;

(* Config with all default values *)
let () =
  let () =
    match Othrottle.Config.t_from_filepath "./config.toml" with
    | Ok config -> Othrottle.Config.sexp_of_t config |> Sexp.to_string |> print_endline
    | Error err -> print_endline err
  in
  ()
;;

(* Config with invalid toml *)
let () =
  let () =
    match
      Othrottle.Config.t_from_filepath "../../../test/samples/invalid_config.toml"
    with
    | Ok config -> Othrottle.Config.sexp_of_t config |> Sexp.to_string |> print_endline
    | Error err -> print_endline err
  in
  ()
;;
