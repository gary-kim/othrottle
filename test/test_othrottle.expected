((shell bash)(job_timeout 30)(task_timeout 1000)(retry_sequence(5 10 30))(retry_on_error true)(notification_cmd"")(filters()))
((shell bash)(job_timeout 600)(task_timeout 30)(retry_sequence(5 15 30 60 120 300 900))(retry_on_error true)(notification_cmd"")(filters()))
Syntax error on line 1, character 14: Malformed key-value pair (missing value?)
